pyexchange/erisx.py

```diff
@@ -0,0 +1,399 @@
+                 fix_marketdata_endpoint: str, fix_marketdata_user: str, password: str,
+                 clearing_url: str, api_key: str, api_secret: str):
+        assert(isinstance(fix_trading_endpoint, str) or (fix_trading_endpoint is None))
+        assert(isinstance(fix_trading_user, str) or (fix_trading_user is None))
+        assert(isinstance(fix_marketdata_endpoint, str) or (fix_marketdata_endpoint is None))
+        assert(isinstance(fix_marketdata_user, str) or (fix_marketdata_user is None))
+        assert(isinstance(password, str) or (password is None))
+        assert(isinstance(clearing_url, str) or (clearing_url is None))
+        assert(isinstance(api_key, str) or (api_key is None))
+        assert(isinstance(api_secret, str) or (api_secret is None))
+
+        # enable access from sync_trades and inventory_service without overriding socket
+        if fix_trading_endpoint is not None and fix_trading_user is not None:
+            self.fix_trading = ErisxFix(fix_trading_endpoint, fix_trading_user, fix_trading_user, password)
+            self.fix_trading.logon()
+            self.fix_trading_user = fix_marketdata_user
+
+        if fix_marketdata_endpoint is not None and fix_marketdata_user is not None:
+            self.fix_marketdata = ErisxFix(fix_marketdata_endpoint, fix_marketdata_user, fix_trading_user, password)
+            self.fix_marketdata.logon()
+            self.fix_marketdata_user = fix_marketdata_user
+
+        self.clearing_url = clearing_url
+        self.api_secret = api_secret
+        self.api_key = api_key
+
+        # store the account id used to retrieve trades and balances
+        self.account_id = self.get_account()
+
+    def __del__(self):
+        self.fix_marketdata.logout()
+        self.fix_trading.logout()
+
+    def ticker(self, pair):
+        # TODO: Subscribe to L1 data, await receipt, and then unsubscribe and return the data.
+        raise NotImplementedError()
+
+    def get_markets(self):
+        # Send 35=x, await 35=y
+        message = self.fix_marketdata.create_message(simplefix.MSGTYPE_SECURITY_LIST_REQUEST)
+        message.append_pair(320, 0)
+        message.append_pair(559, 0)
+        message.append_pair(simplefix.TAG_SYMBOL, 'NA')
+        message.append_pair(460, 2)
+        self.fix_marketdata.write(message)
+        message = self.fix_marketdata.wait_for_response('y')
+        return ErisxFix.parse_security_list(message)
+
+    # filter out the response from get markets
+    def get_pair(self, pair):
+        return self.get_markets()[pair]
+
+    def get_account(self):
+        # Call into the /accounts method of ErisX Clearing WebAPI, which provides a balance of each coin.
+        # They also offer a detailed /balances API, which I don't believe we need at this time.
+        response = self._http_post("accounts", {})
+        if "accounts" in response:
+            return response["accounts"][0]["account_id"]
```

Comment:
I doubt we should place any special importance upon the first account returned by the API.  This impacts the constructor, which calls this method.  Most exchange APIs seem to provide data across all accounts; we could do the same here client-side.

Reply:
are you imaging that as the client would pass an additional argument to the constructor signifying which account they would like to use?

Reply:
I hadn't been looking at it that way because It seemed undesirable to create three new connections for each account we want to trade with.  But it would make this fit into `market-maker-keeper` nicely, which I suppose is the goal.  The `account_id` could be passed as an arg to `market-maker-keeper`, which offers it to the `ErisxApi` constructor.

Reply:
Just updated accordingly with `account_id` arg